Library/Homebrew/dependencies_helpers.rb

```diff
@@ -41,17 +26,14 @@ def recursive_includes(klass, root_dependent, includes, ignores)
     cache_key = "recursive_includes_#{includes}_#{ignores}"
 
     klass.expand(root_dependent, cache_key: cache_key) do |dependent, dep|
-      if dep.recommended?
-        klass.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
-      elsif dep.optional?
-        klass.prune if includes.exclude?("optional?") && !dependent.build.with?(dep)
-      elsif dep.build? || dep.test?
-        keep = false
-        keep ||= dep.test? && includes.include?("test?") && dependent == root_dependent
-        keep ||= dep.build? && includes.include?("build?")
-        klass.prune unless keep
-      elsif dep.satisfied?
-        klass.prune if ignores.include?("satisfied?")
+      klass.prune if ignores.any? { |ignore| dep.send(ignore) }
+
+      # NOTE: Untagged dependencies are runtime dependencies and are included by default.
+      klass.prune if !dep.tags.empty? && includes.none? do |include|
+        # Ignore indirect test dependencies
+        next if include == :test? && dependent != root_dependent
+
+        dep.send(include)
```

Comment:
```suggestion
        dep.public_send(include)
```

assuming it works.

Reply:
Handled in https://github.com/Homebrew/brew/pull/15892/commits/b4be0f088b85949204a365b4c5dd2ec51a4cd0bc